possibilité de supprimer la plage horaire lors de la modification

TimotheeDurand opened the pull request.

TimotheeDurand committed:
possibilité de supprimer la plage horaire lors de la modification

Raul6469 reviewed (dismissed):
Je continuerai sur cette branche pour prendre ça en charge avec le undo-redo

Raul6469 committed:
Undo-redo plage horaire ok

Raul6469 committed:
Pas de commande si valider sans ajouter plage horaire

Si on validait la modification en laissant la plage horaire null alors qu'elle était déjà null, une nouvelle commande était ajoutée à la liste pour rien

AmosTrask commented on project/src/main/java/org/apache/project/controleur/EtatModifierLivraison1.java:
```diff
@@ -16,12 +16,18 @@ public void validerModificationLivraison(Controleur controleur, FenetrePrincipal
+			}else {
+				plageHoraire = new PlageHoraire(heureDeb, heureFin);
+				commandes.ajouteCommande(new CdeModifierLivraison(livraison, heureDeb, heureFin));
+			}
```
le if sert à rien non ?

Raul6469 replied:
Si, il sert, c'est pour le cas où on supprime la plage horaire

TimotheeDurand replied:
Si tu l'enleve, tu crée une plage horaire qui contient des temps null, et ça crash au calcul de tournée.

Pour que la plage horaire soit nulle, il faut que l'attribut plage horaire soit nul.

TimotheeDurand replied:
Attend mais y'a une erreur, j'avais pas écrit ça a la base

TimotheeDurand replied:
Je blame victor

TimotheeDurand replied:
![chrome_2017-11-05_21-42-46](https://user-images.githubusercontent.com/25065175/32418967-edd110c6-c272-11e7-8fa3-57a865318ecf.png)

TimotheeDurand replied:
A corriger

TimotheeDurand committed:
minor refactoring

TimotheeDurand committed:
traiter cas ou plage horaires inchangées (reste vide ET reste a la même valeur)

TimotheeDurand commented:
j'ai traité en plus le cas ou la plage horaire initiale et la plage horaire modifiée sont les mêmes pour pas mettre une commande qui ne fait rien dans la liste de commandes exécitées
